src/prng.rs

```diff
@@ -76,23 +74,51 @@ where
             // Seek to the next chunk of the buffer that encodes an element of F.
             for i in (self.buffer_index..self.buffer.len()).step_by(F::ENCODED_SIZE) {
                 let j = i + F::ENCODED_SIZE;
+
+                if j > self.buffer.len() {
+                    break;
+                }
+
+                self.buffer_index = j;
+
                 if let Some(x) = match F::try_from_random(&self.buffer[i..j]) {
                     Ok(x) => Some(x),
                     Err(FieldError::ModulusOverflow) => None, // reject this sample
                     Err(err) => panic!("unexpected error: {err}"),
                 } {
-                    // Set the buffer index to the next chunk.
-                    self.buffer_index = j;
-                    self.output_written += 1;
                     return x;
                 }
             }
 
-            // Refresh buffer with the next chunk of PRG output.
-            self.seed_stream.fill(&mut self.buffer);
+            // Refresh buffer with the next chunk of PRG output, filling the front of the buffer
+            // with the leftovers. This ensures continuity of the seeds stream after converting the
+            // `Prng` to a new field type via `into_new_field()`.
+            let left_over = self.buffer.len() - self.buffer_index;
+            for i in 0..left_over {
+                self.buffer[i] = self.buffer[self.buffer_index + i];
+            }
+            self.seed_stream.fill(&mut self.buffer[left_over..]);
             self.buffer_index = 0;
         }
     }
+
+    /// Convert this object into a field element generator for a different field.
+    //
+    // TODO(cjpatton) spec: Consider using distinct seeds for distinct field types. Buffering the
+    // seed stream seems to be important for performance, at least according to the benchmarks for
+    // `PrgAes128`. But having to support multiple output types is delicate because the buffer size
+    // is computed from the field modulus.
+    //
+    // TODO(cjpatton) Remove "dead_code" marker after Poplar1 is implemented.
+    #[allow(dead_code)]
```

Comment:
nit: `#[cfg(test)]` would work too, and it'd be harder to forget to remove it because the compiler would force you to do that when you want to use it from a non-test cfg. But this works fine and is going away in the PR stack you've already prepared so it's not worth rebasing for.